### DISCUSSION TITLE
Issue with Uploading Excel Files Larger Than 129 KB in Minio

### DISCUSSION THREAD

**Discussion by cretzuwashere** [minio/minio#18521] [Q&A] [answered]
## Expected Behavior
Files of all sizes, including Excel files larger than 129 KB, should upload without errors.

## Current Behavior
When attempting to upload an Excel file larger than 129 KB, Minio returns an error: "Resource requested is unwritable, please reduce your request rate." No such error occurs for files smaller than 129 KB or other formats.

## Steps to Reproduce (for bugs)
1. Attempt to upload an Excel file of size 129 KB or larger to Minio.
2. Observe the error message.
3. Try uploading a smaller file and note the absence of the error.
4. Repeat steps with varying file sizes to confirm the threshold for the error.

## Context
This issue is preventing the upload of larger files, impacting data management and workflow.

## Regression
No

## Your Environment
Version used (minio --version):  RELEASE.2023-11-15T20-43-25Z (commit-id=38f35463b7fe07fbbe64bb9150d497a755c6206e)
Operating System and version (uname -a): Windows 10
Docker Version: 4.25.1
Runtime: go1.21.4 linux/amd64

**Comment by klauspost** [comment, marked as answer]
Looks like you are using a file system that doesn't work properly.

**Comment by harshavardhana** [comment]
Docker on Windows does not support O_DIRECT or fakes O_DIRECT for bigger blocks - use a Linux host with Linux filesystems if you care about your data.

**Reply by cretzuwashere** [reply]
We've noticed that in our production environment, which uses Linux, everything operates smoothly. 
However, the issue arises in our development setup where we use Docker for Windows. After researching this parameter and conducting tests on various machines belonging to my colleagues, we observed that for those using a similar setup on Windows, the functionality is as expected. 
Any idea how should we proceed?

**Reply by klauspost** [reply]
If you have a MinIO production environment you should get a [subscription](https://min.io/pricing).

You can try `MINIO_API_DISABLE_ODIRECT=on`. No idea if it will help you - but you can try. It could also be https://github.com/minio/minio/pull/18575 - though that shouldn't be triggered on WSL.

**Reply by harshavardhana** [reply]
> However, the issue arises in our development setup where we use Docker for Windows.

Docker for Windows provides an early-stage overlay filesystem that still does not support regular POSIX semantics; this is why native Windows will work fine, and Linux will work fine, of course.

Docker's layering on Windows introduces filesystem layers that are not fully baked and stable from a filesystem point of view.

For the persistence of your critical data, always use Linux systems.
